bullet_train/app/controllers/concerns/controllers/base.rb

```diff
@@ -85,15 +85,14 @@ def enforce_invitation_only
     end
   end
 
-  def set_locale
-    I18n.locale = [
+  def set_locale(&action)
+    locale = [
       current_user&.locale,
       current_user&.current_team&.locale,
       http_accept_language.compatible_language_from(I18n.available_locales),
       I18n.default_locale.to_s
     ].compact.find { |potential_locale| I18n.available_locales.include?(potential_locale.to_sym) }
-    yield
-    I18n.locale = I18n.default_locale
+    I18n.with_locale(locale, &action)
```

Comment:
LGTM :+1:

Should we set `around_action :set_locale` here too as per the docs?

Reply:
🤦‍♂️ Sorry about that, didn't even notice.